### PR TITLE
Pinterest: hide non-content images

### PR DESCRIPTION
**Pull request by stephanfowler**
In the Pinterest overlay, only show article content images. Hide all other images, eg. from commercial components, related content, etc.

pintrest

### CODE REVIEW

File: static/src/javascripts/bootstraps/article.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ define([
     'common/modules/experiments/ab',
     'common/modules/onward/geo-most-popular',
     'common/modules/onward/social-most-popular',
+    'common/modules/social/pintrest',
```
Comment (rich-nguyen) [inline static/src/javascripts/bootstraps/article.js]: `pinterest`

File: static/src/javascripts/projects/common/modules/social/pinterest.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                    $(img).attr('data-pin-nopin', 'true');
+                });
+            });
+        });
+
+        require(['js!https://assets.pinterest.com/js/pinmarklet.js?r=' + new Date().getTime()]);
+    }
+
+    return function () {
+        fastdom.read(function () {
```
Comment (robertberry) [inline static/src/javascripts/projects/common/modules/social/pinterest.js]: You don't actually have to do a `read` when just selecting from the DOM. You only need to do a `read` when you're querying some property of the DOM that will force layout (e.g., 'what are the dimensions of this div?'). Some info on things that trigger layout here: http://gent.ilcore.com/2011/03/how-not-to-trigger-layout-in-webkit.html

File: static/src/javascripts/projects/common/modules/social/pinterest.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+], function (
+    fastdom,
+    bean,
+    $
+) {
+    function launchOverlay(event) {
+        event.preventDefault();
+
+        fastdom.read(function () {
+            $('img:not(.gu-image):not(.responsive-img)').each(function (img) {
```
Comment (robertberry) [inline static/src/javascripts/projects/common/modules/social/pinterest.js]: Same as below comment

**Comment by robertberry** [comment]
:+1:

**Pull request merged**